---
output: github_document
---

<!-- README.md is generated from README.Rmd. Please edit that file -->

```{r, include = FALSE}
knitr::opts_chunk$set(
  collapse = TRUE,
  comment = "#>",
  fig.path = "man/figures/README-",
  out.width = "100%"
)
```

# simpleExpressionProcessing

<!-- badges: start -->
<!-- badges: end -->

The goal of simpleExpressionProcessing is to provide simple tools for downloading and normalizing raw RNA expression data from GEO, including microarray and bulk RNA sequencing experiments, within the `GEOquery` class format. 

## IMPORTANT NOTE

- This is meant to be used for single-dataset `MetaIntegrator` objects
- To fit into a `MetaIntegrator` workflow, datasets should be processed individually, then combined into a multi-dataset object prior to `MetaIntegrator` analysis

## Installation

You can install the development version of simpleExpressionProcessing from [GitHub](https://github.com/) with:

``` r
devtools::install_github("BenSolomon/simpleExpressionProcessing")
```

If difficulty installing dependencies, the `renv` lockfile for the package development environment can by used to install appropriate package versions  prior to installing `simpleExpressionProcessing`

``` r
install.packages("renv")
renv::init() # If not already using renv
lock_path <- "https://raw.githubusercontent.com/BenSolomon/simpleExpressionProcessing/refs/heads/main/renv.lock"
renv::restore(lockfile = lock_path)
```

## Examples

See vignettes and documentation at: https://bensolomon.github.io/simpleExpressionProcessing/

  - [Microarray processing](https://bensolomon.github.io/simpleExpressionProcessing/articles/microarray_analysis.html)
  - [RNAseq processing](https://bensolomon.github.io/simpleExpressionProcessing/articles/rnaseq_analysis.html)

## General workflow description

- GEO data is downloaded with `MetaIntegrator`. This includes an `expr` slot which contains the expression data, as normalized by the data authors
- Raw expression data is downloaded from GEO
  - **Microarray**: Supplemental data files are downloaded with `GEOquery`, such as raw `.CEL` files from the Affymetrix microarray platform
  - **RNAseq**: The gene count matrix generated by NCBI's standardized pipeline, as well as the annotation file used in that pipeline, are downloaded.
- The raw expression files are normalized 
  - **Microarry**: Affymetrix data uses `Affy` package workflow. Agilent data uses `limma` package workflow
  - **RNAseq**: Uses `DESeq2` workflow. Defaults to `vst` normalization, but can also use `rlog` 
- The GEO data object is modified in the following ways:
  - The original `expr` slot is reassigned to the `exprGEO` slot, to indicate it represents the authors' normalized data
  - The raw, non-normalized data from the supplemental files are assigned to the `exprRAW` slot
  - The user normalized data is added to the `expr` slot, which is the slot used by downstream applications using MetaIntegrator GEO objects
- All expression slots (i.e. `expr`, `exprRAW`, and `exprGEO`) are visualized with boxplots to evaluate the normalization process
